pkg/util/kubernetes/kubelet/types_kubelet.go

```diff
@@ -220,14 +220,24 @@ type Conditions struct {
 
 // ContainerStatus contains fields for unmarshalling a Pod.Status.Containers
 type ContainerStatus struct {
-	Name         string         `json:"name"`
-	Image        string         `json:"image"`
-	ImageID      string         `json:"imageID"`
-	ID           string         `json:"containerID"`
-	Ready        bool           `json:"ready"`
-	RestartCount int            `json:"restartCount"`
-	State        ContainerState `json:"state"`
-	LastState    ContainerState `json:"lastState"`
+	Name         string              `json:"name"`
+	Image        string              `json:"image"`
+	ImageID      string              `json:"imageID"`
+	ID           string              `json:"containerID"`
+	Ready        bool                `json:"ready"`
+	RestartCount int                 `json:"restartCount"`
+	State        ContainerState      `json:"state"`
+	LastState    ContainerState      `json:"lastState"`
+	Resources    []ContainerResource `json:"resources,omitempty"`
```

Comment:
I think the name could be more specific, I'm concerned this conflicts with the new `containerStatuses[].resources` field that should be introduced for vertical in-place resizes https://github.com/kubernetes/enhancements/blob/73af5995f336345fbbed140bfb8d6365b2273e0b/keps/sig-node/1287-in-place-update-pod-resources/README.md?plain=1#L245.

Reply:
AllocatedResources would work? It'd also be consistent with the workloadmeta entity attribute

Reply:
yep sounds good